### DISCUSSION TITLE
Address Autocomplete + Embedded Map of Shipping Destination (Checkout Flow)

### DISCUSSION THREAD

**Discussion by nicoestrada** [medusajs/medusa#4403] [Feature Requests]
Reopening this according to [srindom's request](https://github.com/medusajs/medusa/issues/2512):

**Is your feature request related to a problem? Please describe.**
Implementing a shipping address autocomplete feature and adding an embedded map to the order confirmation page. Similar features are found in typical e-commerce platforms to improve conversion rates and other key metrics.

**Describe the solution you'd like**
To optimize the customer journey flow through checkout, an autocomplete feature of shipping location from Google Places API / Lob Address Autocomplete API can save the customer time and keystrokes when confirming an order, thus drastically increasing a store's conversion rate. Furthermore, when a customer does confirm an order, implementing an embedded map of the shipping location can provide a clean and satisfactory interface to ensure with the customer the order will be shipped to the correct location.

**Describe alternatives you've considered**
Perhaps other API's that utilize addresses that are deliverable across the globe

**Additional context**
Implementing these features can also prevent user type errors when confirming an order. Can also increase buyer's confidence in order to increase likelihood of conversions

**Comment by tsvetann** [comment]
Isn't this feature something you can build yourself in your storefront?

**Reply by SalahAdDin** [reply]
To improve the autocomplete feature, an address service would be required, but it is not related to Medusa, yes.
